encoding/thrift/thriftrw-plugin-yarpc/fake_testreporter_test.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"runtime"
+	"runtime/debug"
+
+	"github.com/golang/mock/gomock"
+)
+
+// FakeTestStatus states how a fake TestingT finished.
+type FakeTestStatus int
```

Comment:
We implemented a fake TestingT in an internal repo, satisfying the testify
TestingT interface. gomock's TestReporter is slightly different. But yeah, we
can consolidate the solutions into a single repo somewhere.